### ISSUE TITLE
Import paths for generated type definitions broken

### ISSUE THREAD

**Issue by donmahallem** [remarkjs/remark-lint#270]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-lint-list-item-indent and many more

### Steps to reproduce

Clone repository and run `npm run build` after bootstraping the repository

### Expected behavior

It should generate type defintions like so that type acquisition isn't broken if the dependency is somewhere else.
```
export default remarkLintListItemIndent;
export type Root = import('mdast').Root;
export type Options = 'tab-size' | 'space' | 'mixed';
declare const remarkLintListItemIndent: import("unified").Plugin<void[] | [Options | import("unified-lint-rule").Label | import("unified-lint-rule").Severity | undefined] | [boolean | import("unified-lint-rule").Label | import("unified-lint-rule").Severity, Options | undefined], import("mdast").Root, import("mdast").Root>;
```

### Actual behavior

But it generates which brakes if the unified package isn't a subpackage of unified-lint-rule for example on the same "level"
```
export default remarkLintListItemIndent;
export type Root = import('mdast').Root;
export type Options = 'tab-size' | 'space' | 'mixed';
declare const remarkLintListItemIndent: import("unified-lint-rule/node_modules/unified").Plugin<void[] | [Options | import("unified-lint-rule").Label | import("unified-lint-rule").Severity | undefined] | [boolean | import("unified-lint-rule").Label | import("unified-lint-rule").Severity, Options | undefined], import("mdast").Root, import("mdast").Root>;
```

Additonally it results in many duplicate installs unified etc.
### Runtime

Node v16

### Package manager

npm v7

### OS

Windows

**Comment by donmahallem** [comment]
One solution would be to run `npx lerna bootstrap --hoist` from my testing to remove these nested imports.

**Comment by donmahallem** [comment]
I expected missmatching unified versions BUT it isn't only unified but several remark-lint packages referencing another remark-lint package too. I did clone this repository fresh and it resulted in the same build d.ts files so I suspect no conflicting versions. 

`--hoist` is a hack too imho and should be avoided but it should be indicate that conflicting versions shouldn't be the issue.

**Comment by wooorm** [comment]
The presets went weird becasue they were essentially untyped, hence trying to load tons of external types. I fixed that (see reference above this comment), e.g., they now look like this:

```ts
export default remarkPresetLintRecommended;
export type Preset = import('unified').Preset;
/** @type {Preset} */
declare const remarkPresetLintRecommended: Preset;
```

For rules, they don’t depend on each other. So that should leave the problem to `unified-lint-rule`s `unified` being used instead of the local `unified`s

**Comment by donmahallem** [comment]
Currently away from my PC but I remember imports like

`../[preset]/node_modules/[other dependency]`

Maybe this is fixed with this commit too. I will check this later.

**Comment by wooorm** [comment]
Released!
